feature/gnmi/tests/gnmi_ni_test/gnmi_ni_test.go

```
@@ -55,58 +57,90 @@ func TestCodelab(t *testing.T) {
+	cfgplugins.NewNetworkInstance(t, batch, dut, &dutPort2NetworkInstanceIParams)
 
-	t.Log("\nApplying configuration to DUT\n")
+	// Configure non-default gNMI server.
+	cfgplugins.CreateGNMIServer(batch, t, dut, &dutPort2NetworkInstanceIParams)
+}
 
-	batch.Set(t, dut)
+func ValidateNetworkInstance(t *testing.T, dut *ondatra.DUTDevice) {
+	t.Log("\nValidating network instance after configuration\n")
+	// Get all network instances.
+	netInstanceList := gnmi.GetAll(t, dut, gnmi.OC().NetworkInstanceAny().State())
+	t.Logf("Network instance list length: %v", len(netInstanceList))
+	for _, netInstance := range netInstanceList {
+		t.Logf("----------------------------------------------------")
+		t.Logf("VRF / Network instance")
+		t.Logf("----------------------------------------------------")
+		t.Logf("Network instance name: %s", netInstance.GetName())
+		t.Logf("Network instance type: %s", netInstance.GetType())
 
-	for _, netInstance := range gnmi.GetAll(t, dut, gnmi.OC().NetworkInstanceAny().State()) {
-		t.Logf("Network instance: %s", netInstance.GetName())
 	}
 
 	// Get and validate states for default and custom networkinstances.
 	gnmiServerList := gnmi.GetAll(t, dut, gnmi.OC().System().GrpcServerAny().State())
+	t.Logf("gNMI server list length: %v", len(gnmiServerList))
 	for _, gnmiServer := range gnmiServerList {
-		if gnmiServer.GetName() == deviations.DefaultNetworkInstance(dut) {
-			defaultInstanceState := gnmi.Get(t, dut, gnmi.OC().System().GrpcServer(deviations.DefaultNetworkInstance(dut)).State())
+		t.Logf("----------------------------------------------------")
+		t.Logf("gNMI server name: %v", gnmiServer.GetName())
+		t.Logf("----------------------------------------------------")
+		t.Logf("gNMI server network-instance: %v", gnmiServer.GetNetworkInstance())
+		t.Logf("gNMI server port: %v", gnmiServer.GetPort())
+		t.Logf("gNMI server enable: %v", gnmiServer.GetEnable())
+		t.Logf("gnmi server services: %v", gnmiServer.GetServices())
```

nit: Instead of logging these things, the code could simply verify the expected values and throw a t.Error or t.Fatal is not found.    (Because we don't expect people to be reading logs to determine test pass/fail or get FYI information).   We'd rather see the results of a call to `t.Run()` with some useful test name.  That can be parsed by go test framework/tools.

---

Removed.